apis/keyvault/v1alpha1/keyvault_secret_types.go

```diff
@@ -0,0 +1,183 @@
+	// operation is requested, or the subscription is cancelled.
+	CustomizedRecoverablePurgeable DeletionRecoveryLevel = "CustomizedRecoverable+Purgeable"
+
+	// Purgeable Denotes a vault state in which deletion is an irreversible operation, without the possibility
+	// for recovery. This level corresponds to no protection being available against a Delete operation; the
+	// data is irretrievably lost upon accepting a Delete operation at the entity level or higher (vault,
+	// resource group, subscription etc.)
+	Purgeable DeletionRecoveryLevel = "Purgeable"
+
+	// Recoverable Denotes a vault state in which deletion is recoverable without the possibility for immediate
+	// and permanent deletion (i.e. purge). This level guarantees the recoverability of the deleted entity
+	// during the retention interval(90 days) and while the subscription is still available. System wil
+	// permanently delete it after 90 days, if not recovered
+	Recoverable DeletionRecoveryLevel = "Recoverable"
+
+	// RecoverableProtectedSubscription Denotes a vault and subscription state in which deletion is recoverable
+	// within retention interval (90 days), immediate and permanent deletion (i.e. purge) is not permitted, and
+	// in which the subscription itself  cannot be permanently canceled. System wil permanently delete it after
+	// 90 days, if not recovered
+	RecoverableProtectedSubscription DeletionRecoveryLevel = "Recoverable+ProtectedSubscription"
+
+	// RecoverablePurgeable Denotes a vault state in which deletion is recoverable, and which also permits
+	// immediate and permanent deletion (i.e. purge). This level guarantees the recoverability of the deleted
+	// entity during the retention interval (90 days), unless a Purge operation is requested, or the
+	// subscription is cancelled. System wil permanently delete it after 90 days, if not recovered
+	RecoverablePurgeable DeletionRecoveryLevel = "Recoverable+Purgeable"
+)
+
+// KeyVaultSecretAttributesParameters defines the desired state of an Azure Key Vault Secret Attributes.
+// KeyVaultSecretAttributesParameters contains WRITE-ONLY fields.
+type KeyVaultSecretAttributesParameters struct {
+	// Enabled - Determines whether the object is enabled.
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// NotBefore - Not before date in UTC.
+	NotBefore *metav1.Time `json:"nbf,omitempty"`
+
+	// Expires - Expiry date in UTC.
+	Expires *metav1.Time `json:"exp,omitempty"`
+}
+
+// KeyVaultSecretParameters defines the desired state of an Azure Key Vault Secret.
+// https://docs.microsoft.com/en-us/rest/api/keyvault/#secret-operations
+type KeyVaultSecretParameters struct {
+	// VaultBaseURL - The vault name, for example https://myvault.vault.azure.net.
+	VaultBaseURL string `json:"vaultBaseUrl"`
+
+	// Name - The name of the secret
+	Name string `json:"name"`
+
+	// Value - The value of the secret
+	Value string `json:"value"`
```

Comment:
Maybe we should use a secretRef instead of a plain/text string field. As we are handling a sensitive information in this field.